GeneralJS/balanced-parens-2.ts

```diff
@@ -16,6 +16,91 @@
  *
  */
 
-function balancedParens2(input): boolean {}
+// solve parens 1 first lol
+
+function balancedParens2(input) {
+    /*
+        ((())[]{}) = true
+        var hubble = function() { telescopes.awesome();) = false
+    */
+   
+    // breaking case : closed symbol is proceeded by a none matching open symbol
+
+    // define regex
+    const parsingRegex = /[\[\]\{\}\(\)]/g;
+    const openRegex = /[\[\{\(]/g
+    const closedRegex = /[\]\}\)]/g
+    // parens only
+    const parsedInput = input.match(parsingRegex)
+    // keep up with pairs 
+    let parenStack = []; 
+
+    // if there are no parens
+    if(parsedInput == null) return true
+    // if we start with an open paren
+    if(parsedInput[0].match(openRegex) == null) return false;
+
+    // iterate over parens
+    for(let i = 0; i < parsedInput.length; i++) {
+        // no one wants to type this every time
+        let curr = parsedInput[i]
+        let prev = parsedInput[i - 1]
+
+        // check order to remedy failing test : [(]{)}
+        if(curr.match(closedRegex)){
+            if(curr == ')' && (prev == '{' || prev == '[')) return false
+            if(curr == ']' && (prev == '{' || prev == '(')) return false
+            if(curr == '}' && (prev == '(' || prev == '[')) return false
+        }
+
+        // equal matches
+        if(curr.match(openRegex)){
+            parenStack.push(curr)
+        }
+        if(curr.match(closedRegex)){
+            parenStack.pop();
+        }
```

Comment:
Looking good!
For further simplification, how can we smoosh this into one if/else?

Hint 1: you don't need to keep track of prev (you're already keeping track of the values you need in your stack
Hint 2: it's good to know what's open vs what's closed, and it is also useful to know what matches what